src/foundry/client/pixi/placeables/primary-canvas-objects/occludable-object.d.ts

```diff
@@ -0,0 +1,167 @@
+   */
+  updateTextureData(): { minX: number; minY: number; maxX: number; maxY: number; pixels: Uint8Array | undefined };
+
+  /**
+   * Test whether a specific Token occludes this PCO.
+   * Occlusion is tested against 9 points, the center, the four corners-, and the four cardinal directions
+   * @param token   - The Token to test
+   * @param options - Additional options that affect testing
+   * @returns Is the Token occluded by the PCO?
+   */
+  testOcclusion(
+    token: Token,
+    options?: {
+      /** Test corners of the hit-box in addition to the token center? */
+      corners?: boolean;
+    },
+  ): boolean;
+
+  /**
+   * Test whether the PCO pixel data contains a specific point in canvas space
+   * @param alphaThreshold - Value from which the pixel is taken into account, in the range [0, 1].
+   *                         (default: `0.75`)
+   */
+  containsPixel(x: number, y: number, alphaThreshold?: number): boolean;
+
+  /**
+   * Get alpha value at specific canvas coordinate.
+   * @returns The alpha value (-1 if outside of the bounds) or null if no mesh or texture is present.
+   */
+  getPixelAlpha(x: number, y: number): number | null;
+
+  /**
+   * Get PCO alpha map texture coordinate with canvas coordinate
+   * @param testX - Canvas x coordinate.
+   * @param testY - Canvas y coordinate.
+   * @returns The texture `{x, y}` coordinates, or null if not able to do the conversion.
+   */
+  protected _getTextureCoordinate(testX: number, testY: number): { x: number; y: number } | null;
+
+  /**
+   * Compute the alpha-based bounding box for the tile, including an angle of rotation.
+   * @internal
+   */
+  protected _getAlphaBounds(): PIXI.Rectangle;
+
+  /**
+   * @deprecated since v11, will be removed in v13
+   * @remarks "PrimaryCanvasObject#renderOcclusion is deprecated in favor of PrimaryCanvasObject#renderDepth"
+   */
+  renderOcclusion(renderer: PIXI.Renderer): void;
+}
+
+declare global {
+  namespace OccludableObject {
+    type Data = {
+      /**
+       * The PCO is considered as a roof?
+       * @defaultValue false
+       */
+      roof: boolean;
+
+      /** The occlusion object for this PCO */
+      occlusion: {
+        /** @defaultValue `CONST.OCCLUSION_MODES.NONE` */
+        mode: OCCLUSION_MODES;
+        /** @defaultValue `0` */
+        alpha: number;
+        /** @defaultValue `null` */
+        radius: number | null;
+      };
+    } & PrimaryCanvasObject.Data;
+  }
+
+  /**
+   * A mixin which decorates a DisplayObject with depth and/or occlusion properties.
+   * @param DisplayObject - The parent DisplayObject class being mixed
+   * @returns A DisplayObject subclass mixed with OccludableObject features
+   */
+  function OccludableObjectMixin<BaseClass extends typeof SpriteMesh>(
+    DisplayObject: BaseClass,
+  ): Mixin<typeof OccludableObject, BaseClass> & ReturnType<typeof PrimaryCanvasObjectMixin<BaseClass>>;
```

Comment:
Judging off of feelings it should probably be `Mixin<typeof OccludableObject, ReturnType<typeof PrimaryCanvasObjectMixin<BaseClass>>>;`. Can you post the source so I can take a look?

Reply:
Here it is
```js
/**
 * A mixin which decorates a DisplayObject with depth and/or occlusion properties.
 * @category - Mixins
 * @param {typeof PIXI.DisplayObject} DisplayObject   The parent DisplayObject class being mixed
 * @returns {typeof OccludableObject}                 A DisplayObject subclass mixed with OccludableObject features
 */
function OccludableObjectMixin(DisplayObject) {
  // Verify that the display object is a prototype of SpriteMesh (for occlusion, we need the shader class support)
  // TODO: Remove PIXI.TilingSprite as soon as possible!
  if ( !(foundry.utils.isSubclass(DisplayObject, SpriteMesh) ||
    foundry.utils.isSubclass(DisplayObject, PIXI.TilingSprite)) ) {
    throw new Error("Occludable objects must be a subclass of SpriteMesh.");
  }
  return class OccludableObject extends PrimaryCanvasObjectMixin(DisplayObject) {}
}
```

Reply:
Yeah, it should be `Mixin<typeof OccluableObject, ReturnType<typeof PrimaryCanvasObjectMixin<BaseClass>>`

Does it make sense to you why that is? If I had to briefly explain it's basically because it really is a mixin nesting a mixin like so.